phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -5079,6 +5079,38 @@ private void incrementSequenceValues(List<SequenceAllocation> sequenceAllocation
         }
     }
 
+    /**
+     * checks if sequenceAllocation's sequence there in sequenceMap, also returns Global Sequences
+     * from Tenant sequenceAllocations
+     * @param sequenceAllocation
+     * @return
+     */
+
+    private Sequence getSequence(SequenceAllocation sequenceAllocation) {
+        SequenceKey key = sequenceAllocation.getSequenceKey();
+        if (key.getTenantId() == null) {
+            return sequenceMap.putIfAbsent(key, new Sequence(key));
+        } else {
+            Sequence sequence = sequenceMap.get(key);
+            if (sequence == null) {
+                for (Map.Entry<SequenceKey,Sequence> entry : sequenceMap.entrySet()) {
+                    if (compareSequenceKeysWithoutTenant(key, entry.getKey())) {
+                        return entry.getValue();
+                    }
+                }
+            } else {
+                return sequence;
+            }
+        }
+        return null;
+    }
+
+    private boolean compareSequenceKeysWithoutTenant(SequenceKey key1, SequenceKey key2) {
+        return key2.getTenantId() == null && (key1.getSchemaName() == null ? key2.getSchemaName() == null :
+                key1.getSchemaName().equals(key2.getSchemaName())) &&
+                key1.getSequenceName().equals(key2.getSequenceName());
```

Comment:
Here we are checking if key2 is global sequence right?

It's bit difficult to understand in one statement. Good to break it down, something like
```
    private boolean compareSequenceKeysWithoutTenant(SequenceKey keyToCompare,
            SequenceKey availableKey) {
        boolean sameSchema = keyToCompare.getSchemaName() == null
                ? availableKey.getSchemaName() == null
                : keyToCompare.getSchemaName().equals(availableKey.getSchemaName());
        if (!sameSchema) {
            return false;
        }
        if (availableKey.getTenantId() != null) {
            return false;
        }
        return keyToCompare.getSequenceName().equals(availableKey.getSequenceName());
    }
```